src/include/common/copier_config/reader_config.h

```diff
@@ -0,0 +1,45 @@
+#pragma once
+
+#include <string>
+#include <vector>
+
+namespace kuzu {
+namespace common {
+
+enum class FileType : uint8_t { UNKNOWN = 0, CSV = 1, PARQUET = 2, NPY = 3, TURTLE = 4 };
+
+struct FileTypeUtils {
+    static FileType getFileTypeFromExtension(const std::string& extension);
+    static std::string toString(FileType fileType);
+};
+
+struct ExtraReaderConfig {
+    virtual ~ExtraReaderConfig() = default;
+    virtual std::unique_ptr<ExtraReaderConfig> copy() = 0;
+};
+
+struct ReaderConfig {
+    FileType fileType = FileType::UNKNOWN;
+    std::vector<std::string> filePaths;
```

Comment:
FilePath is not needed when scanning from pandas/numpy, so move this field to child class?